utils/srcmanip/change_header_year

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+import os
+import re
+import fnmatch
+import argparse
+
+
+DESCRIPTION = "Replaces the year in files with existing copyright header."
+
+YEAR = "2024"
+
+PATTERN = re.compile(
+    r"""Copyright\s+\(C\)\s+(\d{4}\s*-\s*)*(?P<year>\d{4})\s+DFTB\+\s+developers\s+group"""
+)
+
+FOLDERS = ["app/", "cmake/", "src/", "sys/", "test/", "tools/", "utils/"]
+
+FILES = ["LICENSE"]
+
+CHANGE_MANUALLY = ["src/dftbp/common/release.F90", "doc/dptools/api/conf.py"]
+
+IGNORED_FOLDERS = ["__pycache__"]
+
+IGNORED_FILES = [
+    "*~",
+    "*.pyc",
+    "*/make.deps",
+    "*/make.extdeps",
+    "*.mgf",
+    "*.egg",
+    "*.skf",
+]
+
+
+def main():
+    """Main function"""
+    args = parse_arguments()
+    rootdir = os.path.abspath(args.rootdir)
+    process_files(rootdir)
+    print_files_manually(rootdir)
+
+
+def parse_arguments():
+    """Parses the arguments"""
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    msg = "Root directory of the project (default: current directory)"
+    parser.add_argument("--rootdir", default=os.getcwd(), help=msg)
+    args = parser.parse_args()
+    return args
+
+
+def process_files(rootdir):
+    """Replaces the year in header"""
+    file_list = find_files(rootdir)
+    number_files = len(file_list)
+    indent = len(str(number_files)) + 1
+    for num, file in enumerate(file_list):
+        changed_date = False
+        with open(file, "r", encoding="utf-8") as fp:
+            newcontent = ""
+            for line in fp.readlines():
+                hit = PATTERN.search(line)
+                if hit:
+                    changed_date = True
+                    start, stop = hit.span("year")
+                    newcontent += line[:start] + YEAR + line[stop:]
+                else:
+                    newcontent += line
```

Comment:
Is there any reason, why we have to process the file line by line (as instead of reading in one go and make a search and replace)?